#! /bin/sh -e

# DP: fix/hack to build 32bit libjava on amd64.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- libjava/include/i386-signal.h~	2004-09-26 20:38:48.000000000 +0000
+++ libjava/include/i386-signal.h	2005-06-01 18:16:15.107898552 +0000
@@ -19,6 +19,13 @@
 #define HANDLE_SEGV 1
 #define HANDLE_FPE 1
 
+#ifndef SYS_sigaction
+#define SYS_sigaction SYS_rt_sigaction
+#endif
+#ifndef __NR_sigreturn
+#define __NR_sigreturn __NR_rt_sigreturn
+#endif
+
 #define SIGNAL_HANDLER(_name)	\
 static void _name (int _dummy __attribute__ ((__unused__)))
 
